Mathlib/SetTheory/Ordinal/Basic.lean

```diff
@@ -1065,46 +1065,40 @@ theorem mk_toType (o : Ordinal) : #o.toType = o.card :=
 @[deprecated mk_toType (since := "2024-08-26")]
 alias mk_ordinal_out := mk_toType
 
-/-- The ordinal corresponding to a cardinal `c` is the least ordinal
-  whose cardinal is `c`. For the order-embedding version, see `ord.order_embedding`. -/
-def ord (c : Cardinal) : Ordinal :=
-  let F := fun α : Type u => ⨅ r : { r // IsWellOrder α r }, @type α r.1 r.2
-  Quot.liftOn c F
-    (by
-      suffices ∀ {α β}, α ≈ β → F α ≤ F β from
-        fun α β h => (this h).antisymm (this (Setoid.symm h))
-      rintro α β ⟨f⟩
-      refine le_ciInf_iff'.2 fun i => ?_
-      haveI := @RelEmbedding.isWellOrder _ _ (f ⁻¹'o i.1) _ (↑(RelIso.preimage f i.1)) i.2
-      exact
-        (ciInf_le' _
-              (Subtype.mk (f ⁻¹'o i.val)
-                (@RelEmbedding.isWellOrder _ _ _ _ (↑(RelIso.preimage f i.1)) i.2))).trans_eq
-          (Quot.sound ⟨RelIso.preimage f i.1⟩))
-
+/-- The ordinal corresponding to a cardinal `c` is the least ordinal whose cardinal is `c`. For the
+order-embedding version, see `ord.orderEmbedding`. -/
+def ord (c : Cardinal) : Ordinal := by
+  apply Quot.liftOn c fun α : Type u => ⨅ r : { r // IsWellOrder α r }, @type α r.1 r.2
```

Comment:
Could you please move the data building part out of `by`? AFAIR, @kim-em was trying to make `Mathlib` compile with each `by` replaced with `by sorry`.